lib/MusicBrainz/Server/Data/ArtistCredit.pm

```diff
@@ -25,26 +27,31 @@ sub get_by_ids
                 "JOIN artist ON artist.id=artist_credit_name.artist " .
                 "WHERE artist_credit IN (" . placeholders(@ids) . ") " .
                 "ORDER BY artist_credit, position";
+
     my %result;
     my %counts;
-    foreach my $id (@ids) {
-        my $obj = MusicBrainz::Server::Entity::ArtistCredit->new(id => $id);
-        $result{$id} = $obj;
-        $counts{$id} = 0;
-    }
     for my $row (@{ $self->sql->select_list_of_hashes($query, @ids) }) {
-        my %info = (
-            artist_id => $row->{artist},
-            name => $row->{ac_name}
-        );
-        $info{join_phrase} = $row->{join_phrase} // '';
-        my $obj = MusicBrainz::Server::Entity::ArtistCreditName->new(%info);
-        $obj->artist($self->c->model('Artist')->_new_from_row($row));
         my $id = $row->{artist_credit};
-        $result{$id}->add_name($obj);
+        $counts{$id} //= 0;
+        $result{$id} //= MusicBrainz::Server::Entity::ArtistCredit->new(id => $id);
```

Comment:
This is now inside the loop processing the database results, so the object is only created if the AC actually exists in the database.